sips/sip-012/sip-012-cost-limits-network-upgrade.md

```diff
@@ -0,0 +1,742 @@
+also lower than the theoretical maximum one would expect. On the other hand, the
+mempool consistently has hundreds of valid transactions pending at any given
+time; at peak there have been several thousand pending transactions. So what is
+preventing more transactions from being included in blocks?
+
+The answer is cost-limits. An analysis of “full blocks” (meaning blocks where at
+least one cost dimension exceeds the block limit -- see SIP-006 for a
+description of all cost dimensions) found that roughly 72% of those blocks hit
+the `runtime` limit and ~27% hit the `read_count` limit. Another analysis of all
+transactions rejected due to cost limits found that 90% of those transactions
+exceeded the `runtime` limit. So the `runtime` limits are the primary bottleneck
+right now.
+
+In the last few months, the
+[clarity-benchmarking](https://github.com/blockstack/clarity-benchmarking)
+project has done rigorous benchmarking on contemporary hardware to come up with
+more accurate cost-limits, with a focus on the `runtime` limits. The updated
+cost-limits are described in detail in [this forum
+post](https://forum.stacks.org/t/more-accurate-cost-functions-for-clarity-native-functions/12386).
+
+Any modification of cost-limits is a consensus-breaking change. We believe that
+there is broad community support for changing the cost-limits, the question is exactly
+how and when they go into effect. A previous proposal suggested using a voting
+contract to determine the block height at which a network-upgrade, described in
+detail in [this Github
+discussion](https://github.com/blockstack/stacks-blockchain/discussions/2845).
+Unfortunately, this path would take at least 4 months in the best-case scenario.
+
+This SIP posits that the ongoing network congestion warrants a more expedient
+route to change the cost-limits, one that does not rely on an on-chain voting
+contract. Equally of note, we consider this SIP as a method of last resort but
+that considering the circumstances an exception is justified. All future network
+upgrades should use the voting contract (if appropriate); all hard-forks must
+follow the processes described in SIP-000 and SIP-011.
+
+
+# Specification
+
+## Assumptions
+
+
+
+## Proposal
+
+The Stacks Foundation or the governance group should choose a Bitcoin block
+height for the network upgrade. The block number should be at least two calendar
+weeks from when this SIP transitions into “Accepted” state, so as to provide
+sufficient time for node operators to upgrade.
+
+Miners, developers, Stackers and community members can demonstrate their support
+for this network upgrade in one of the following two ways:
+
+* _Send a contract-call transaction to indicate support for the upgrade_: The
+  contract would aggregate the STX balance on the `tx-sender` account. It could
+  additionally call into the PoX contract to separately aggregate the total
+  Stacked STX amount. See the Appendix for an initial draft of such a contract.
```

Comment:
What is the role of the initial draft here? Does it have any meaning for the SIP or the activation process? It suggests a certain solution without implying any. I mean the final implementation could be different, couldn't it?